bundles/org.openhab.binding.vigicrues/src/main/java/org/openhab/binding/vigicrues/internal/handler/VigiCruesHandler.java

```diff
@@ -102,26 +211,42 @@ public void handleCommand(ChannelUID channelUID, Command command) {
     }
 
     private void updateAndPublish() {
+        StationConfiguration config = getConfigAs(StationConfiguration.class);
         try {
-            if (queryUrl != null) {
-                String response = HttpUtil.executeUrl("GET", queryUrl, TIMEOUT_MS);
-                updateStatus(ThingStatus.ONLINE);
-                OpenDatasoftResponse apiResponse = gson.fromJson(response, OpenDatasoftResponse.class);
-                Arrays.stream(apiResponse.getRecords()).findFirst().flatMap(Record::getFields).ifPresent(field -> {
-                    field.getHeight().ifPresent(height -> updateQuantity(HEIGHT, height, METRE));
-                    field.getFlow().ifPresent(flow -> updateQuantity(FLOW, flow, CUBICMETRE_PER_SECOND));
-                    field.getTimestamp().ifPresent(date -> updateDate(OBSERVATION_TIME, date));
+            OpenDatasoftResponse measures = apiHandler.getMeasures(config.id);
+            measures.getFirstRecord().ifPresent(field -> {
+                field.getHeight().ifPresent(height -> {
+                    updateQuantity(HEIGHT, height, SIUnits.METRE);
+                    updateRelativeMeasure(RELATIVE_HEIGHT, referenceHeights, height);
                 });
-            } else {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.DISABLED,
-                        "queryUrl should never be null, but it is !");
+                field.getFlow().ifPresent(flow -> {
+                    updateQuantity(FLOW, flow, SmartHomeUnits.CUBICMETRE_PER_SECOND);
+                    updateRelativeMeasure(RELATIVE_FLOW, referenceFlows, flow);
+                });
+                field.getTimestamp().ifPresent(date -> updateDate(OBSERVATION_TIME, date));
+            });
+            if (portion != null) {
+                InfoVigiCru status = apiHandler.getTronconStatus(portion);
+                updateAlert(ALERT, status.vicInfoVigiCru.vicNivInfoVigiCru - 1);
+                updateString(SHORT_COMMENT, status.vicInfoVigiCru.vicSituActuInfoVigiCru);
+                updateString(COMMENT, status.vicInfoVigiCru.vicQualifInfoVigiCru);
             }
-        } catch (MalformedURLException e) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                    String.format("Querying '%s' raised : %s", queryUrl, e.getMessage()));
-        } catch (IOException e) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                    String.format("Error opening connection to VigiCrues webservice : {}", e.getMessage()));
+            updateStatus(ThingStatus.ONLINE);
+        } catch (VigiCruesException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.DISABLED, e.getMessage());
```

Comment:
Bindings should never use `ThingStatusDetail.DISABLED`, that is reserved for the core. Please use `ThingStatusDetail.COMMUNICATION_ERROR` instead.